packages/@aws-cdk/assets/lib/fs/fingerprint.ts

```diff
@@ -38,40 +38,28 @@ export function fingerprint(fileOrDirectory: string, options: FingerprintOptions
   const rootDirectory = fs.statSync(fileOrDirectory).isDirectory()
     ? fileOrDirectory
     : path.dirname(fileOrDirectory);
-  const exclude = options.exclude || [];
-  _processFileOrDirectory(fileOrDirectory);
 
-  return hash.digest('hex');
-
-  function _processFileOrDirectory(symbolicPath: string, realPath = symbolicPath) {
-    if (shouldExclude(exclude, symbolicPath)) {
-      return;
-    }
-
-    const stat = fs.lstatSync(realPath);
-    const relativePath = path.relative(fileOrDirectory, symbolicPath);
+  for (const assetFile of listFilesRecursively(fileOrDirectory, {...options, follow}, rootDirectory)) {
+    const relativePath = path.relative(fileOrDirectory, assetFile.absolutePath);
```

Comment:
Can you not use `relativePath` from the AssetFile here?

Reply:
It is relative to `fileOrDirectory` instead of `rootDirectory`